git-service/src/softserve/handlers.rs

```diff
@@ -1,32 +1,103 @@
-};
-use super::{create_token, create_user};
-use actix_web::{web, HttpResponse, Responder};
-
-pub async fn handle_create_user(user: web::Json<CreateUserRequest>) -> impl Responder {
-    match create_user(&user.username) {
-        Ok(message) => HttpResponse::Ok().json(CreateUserResponse {
-            username: user.username.clone(),
-            message,
-        }),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Error: {}", e)),
+use super::errors::GitServiceError;
+use super::models::{CreateRepoRequest, CreateTokenRequest, CreateUserRequest};
+use super::ssh::execute_command;
+use super::{create_repo, create_token, create_user};
+use actix_web::{web, HttpResponse, Result};
+use serde_json::json;
+use tracing::{error, info, warn};
+
+pub async fn handle_create_user(
+    user: web::Json<CreateUserRequest>,
+) -> Result<HttpResponse, GitServiceError> {
+    if check_user_exists(&user.username)? {
+        warn!("Attempt to create existing user: {}", user.username);
+        return Err(GitServiceError::UserAlreadyExists(user.username.clone()));
     }
+
+    create_user(&user.username)
+        .map(|response| {
+            info!("User created: {}", user.username);
+            Ok(HttpResponse::Ok().json(response))
+        })
+        .map_err(|e| {
+            error!("Failed to create user {}: {}", user.username, e);
+            GitServiceError::FailedToCreateUser(e.to_string())
+        })?
 }
 
-pub async fn handle_create_token(token_req: web::Json<CreateTokenRequest>) -> impl Responder {
-    match create_token(&token_req.token_name) {
-        Ok(token) => HttpResponse::Ok().json(CreateTokenResponse {
-            token_name: token_req.token_name.clone(),
-            token: token.trim().to_string(),
-        }),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Error: {}", e)),
-    }
+pub async fn handle_create_token(
+    token_req: web::Json<CreateTokenRequest>,
+) -> Result<HttpResponse, GitServiceError> {
+    create_token(&token_req.token_name)
+        .map(|response| {
+            info!("Token created: {}", token_req.token_name);
+            Ok(HttpResponse::Ok().json(response))
+        })
+        .map_err(|e| {
+            error!("Failed to create token {}: {}", token_req.token_name, e);
+            GitServiceError::FailedToCreateToken(e.to_string())
+        })?
 }
 
-pub async fn test_connection() -> impl Responder {
-    match create_user("test_user") {
-        Ok(_) => HttpResponse::Ok().body("Connection to Soft Serve successful"),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Connection failed: {}", e)),
+pub fn check_repo_exists(repo_name: &str) -> Result<bool, GitServiceError> {
+    let command = "repo list";
+    execute_command(command)
+        .map(|output| {
+            let repos: Vec<&str> = output.lines().collect();
+            repos.iter().any(|&repo| repo.trim() == repo_name)
```

Comment:
You may want to convert the repo_name to lowercase before validation. We should do the same when saving the repo name in the database
```suggestion
            repos.iter().any(|&repo| repo.to_lowercase().trim() == repo_name.trim().to_lowercase())
```